lib/utilities/will-interrupt-process.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+// work around misbehaving libraries, so we can correctly cleanup before actually exiting.
+//
+// @todo: https://github.com/ember-cli/capture-exit/pull/18
+var exit = require('capture-exit');
+
+var windowsCtrlCTrap;
+var handlers = [];
+
+module.exports = {
+  /**
+   * Add process interruption handler
+   *
+   * When the first handler is added then automatically
+   * sets up process interruption signals listeners
+   *
+   * @public
+   * @method addHandler
+   * @param {function} cb   Callback to be called when process interruption fired
+   */
+  addHandler: function(cb) {
+    var index = handlers.indexOf(cb);
+    if (index > -1) { return; }
+
+    if (!handlers.length) {
+      setupSignalsTrap();
+    }
+
+    handlers.push(cb);
+    exit.onExit(cb);
+  },
+
+  /**
+   * Remove process interruption handler
+   *
+   * If there are no remaining handlers after removal
+   * then clean up all the process interruption signal listeners
+   *
+   * @public
+   * @method removeHandler
+   * @param {function} cb   Callback to be removed
+   */
+  removeHandler: function(cb) {
+    var index = handlers.indexOf(cb);
+    if (index < 0) { return; }
+
+    handlers.splice(index, 1);
+    exit.offExit(cb);
+
+    if (!handlers.length) {
+      teardownSignalsTrap();
+    }
+  },
+
+  /**
+   * Tears down all the handlers and cleans up all signal listeners
+   *
+   * @public
+   * @method teardown
+   */
+  teardown: function() {
+    while (handlers.length) {
+      exit.offExit(handlers.pop());
+    }
+
+    teardownSignalsTrap();
+  }
+};
+
+/**
+ * @public
+ * @method setup
+ */
```

Comment:
these yuidocs don't look correctly any more; i think they're leftover from when this code was moved